src/main/java/com/sprint/mission/discodeit/service/basic/BasicUserService.java

```diff
@@ -0,0 +1,71 @@
+package com.sprint.mission.discodeit.service.basic;
+
+import com.sprint.mission.discodeit.dto.entity.User;
+import com.sprint.mission.discodeit.dto.entity.UserStatus;
+import com.sprint.mission.discodeit.dto.form.CheckUserDto;
+import com.sprint.mission.discodeit.dto.form.UserUpdateDto;
+import com.sprint.mission.discodeit.repository.UserRepository;
+import com.sprint.mission.discodeit.service.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class BasicUserService implements UserService {
+    private final UserRepository userRepository;
+
+    @Override
+    public void createUser(User user) {
+        List<User> all = userRepository.findAll();
+        for (User findUser : all) {
+            if (user.getUserName().equals(findUser.getUserName())) {
+                log.info("중복된 이름입니다.");
+                return;
+            }
+            if (user.getUserEmail().equals(findUser.getUserEmail())) {
+                log.info("중복된 이메일입니다.");
+                return;
+            }
+        }
+        userRepository.createUser(user.getId(), user);
+        UserStatus userStatus = new UserStatus(user.getId());
+    }
+
+    @Override
+    public Optional<CheckUserDto> findUser(UUID id) {
+        return userRepository.findById(id).map(CheckUserDto::new);
+    }
+
+    @Override
+    public Optional<CheckUserDto> findByloginId(String loginId) {
+        return userRepository.findByloginId(loginId).map(CheckUserDto::new);
+    }
+
+    @Override
+    public List<User> findAllUsers() {
+        return userRepository.findAll();
+    }
+
+    @Override
+    public void updateUser(UUID id, UserUpdateDto userParam) {
+        validateUserExits(id);
+        userRepository.updateUser(id, userParam);
+    }
+
+    @Override
+    public void deleteUser(UUID id) {
+        validateUserExits(id);
+        userRepository.deleteUser(id);
+    }
+    private void validateUserExits(UUID uuid) {
+        if (!userRepository.findById(uuid).isPresent()) {
```

Comment:
Optional.isEmpty() 메서드를 사용하면 코드가 더 간결해질 것 같아요!

```suggestion
        if (userRepository.findById(uuid).isEmpty()) {
```